### PR TITLE
Implement XPending redis command

### LINKED ISSUE #34411: Redis StreamCommands should provide xpending command
### Description

Today the only way to request pending messages is to use low level RedisAPI.

### Implementation ideas

_No response_

### PR DESCRIPTION
**Pull request by cescoffier**
Fix #34411.

### CODE REVIEW

File: extensions/redis-client/runtime/src/main/java/io/quarkus/redis/runtime/datasource/ReactiveTransactionalStreamCommandsImpl.java [hunk 1]
```diff
@@ -219,4 +220,22 @@ public Uni<Void> xtrim(K key, XTrimArgs args) {
...
+
+    @Override
+    public Uni<Void> xpending(K key, String group, StreamRange range, int count) {
+        this.tx.enqueue(reactive::decodeListOfPendingMessages);
+        return this.reactive._xpending(key, group, range, count, null).invoke(this::queuedOrDiscard).replaceWithVoid();
+    }
+
+    @Override
+    public Uni<Void> xpending(K key, String group, StreamRange range, int count, XPendingArgs args) {
+        this.tx.enqueue(reactive::decodeAsXPendingSummary);
```
Comment (machi1990) [inline extensions/redis-client/runtime/src/main/java/io/quarkus/redis/runtime/datasource/ReactiveTransactionalStreamCommandsImpl.java:238]: should this be 
```suggestion
        this.tx.enqueue(reactive::decodeListOfPendingMessages);
```

**Review by machi1990 — CHANGES_REQUESTED** [review]
the code change lgtm,thanks @cescoffier. I've however one question/suggestion. Let me know what you think @cescoffier

File: extensions/redis-client/runtime/src/main/java/io/quarkus/redis/runtime/datasource/ReactiveTransactionalStreamCommandsImpl.java (hunk 1 above)
Comment (cescoffier) [reply extensions/redis-client/runtime/src/main/java/io/quarkus/redis/runtime/datasource/ReactiveTransactionalStreamCommandsImpl.java:238]: Oh good catch! 
My transaction test didn't use that method....

File: extensions/redis-client/runtime/src/test/java/io/quarkus/redis/datasource/TransactionalStreamCommandsTest.java [hunk 2]
```diff
@@ -5,14 +5,13 @@
...
+import java.util.stream.Collectors;
 
 import org.junit.jupiter.api.AfterEach;
 import org.junit.jupiter.api.BeforeEach;
 import org.junit.jupiter.api.Test;
 
-import io.quarkus.redis.datasource.stream.StreamMessage;
-import io.quarkus.redis.datasource.stream.TransactionalStreamCommands;
-import io.quarkus.redis.datasource.stream.XAddArgs;
+import io.quarkus.redis.datasource.stream.*;
```
Comment (machi1990) [inline extensions/redis-client/runtime/src/test/java/io/quarkus/redis/datasource/TransactionalStreamCommandsTest.java:14]: nit: wildcard imports

**Review by machi1990 — APPROVED** [review]
lgtm, thanks @cescoffier 

I left two suggestions.

File: extensions/redis-client/runtime/src/main/java/io/quarkus/redis/runtime/datasource/ReactiveTransactionalStreamCommandsImpl.java [hunk 3]
```diff
@@ -219,4 +220,22 @@ public Uni<Void> xtrim(K key, XTrimArgs args) {
...
+    public Uni<Void> xpending(K key, String group) {
+        this.tx.enqueue(reactive::decodeAsXPendingSummary);
+        return this.reactive._xpending(key, group).invoke(this::queuedOrDiscard).replaceWithVoid();
+    }
+
+    @Override
+    public Uni<Void> xpending(K key, String group, StreamRange range, int count) {
+        this.tx.enqueue(reactive::decodeListOfPendingMessages);
+        return this.reactive._xpending(key, group, range, count, null).invoke(this::queuedOrDiscard).replaceWithVoid();
+    }
```
[the remark this answers is not in the document]
Comment (cescoffier) [reply extensions/redis-client/runtime/src/main/java/io/quarkus/redis/runtime/datasource/ReactiveTransactionalStreamCommandsImpl.java:234]: Yes... my tooling is not smart enough to generate the nice one.

**Pull request merged**